src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -587,6 +592,19 @@ MulticopterPositionControl::task_main()
 		// set dt for control blocks
 		setDt(dt);
 
+		// activate the weathervane controller if required. If activated a flighttask can use it to implement a yaw-rate control strategy
+		// that turns the nose of the vehicle into the wind
+		if (_control_mode.flag_control_manual_enabled && _control_mode.flag_control_attitude_enabled
+		    && _wv_controller.manual_enabled()) {
+			_wv_controller.activate();
+
+		} else if (_control_mode.flag_control_auto_enabled && _wv_controller.auto_enabled()) {
+			_wv_controller.activate();
```

Comment:
there is not difference between auto_enabled and manual_enabled? why do these checks then even exist?

Reply:
@Stifael There is a difference. You can have weathervane enabled for manual mode or for auto modes via parameter.

Reply:
What I don't really understand is how the distinction is made between vtol and normal quad. So here you activate / deactivate based on control-mode flags. In FlightTaskAuto and FlightTaskStabilzed you then just check if the controller is active (https://github.com/PX4/Firmware/pull/10411/files#diff-0333660f3bde6b05a591e33140d9753bR92), and if yes, then you will use it.

Reply:
@Stifael At the moment there is none, but we could add that if you feel.like it's necessary. In theory this feature is not limitted to vtol. However, I agree that there is probably not a big use case for multirotors.

Reply:
Of course you need to enable wethervane with the dedicated parameters, it won't just be active always.

Reply:
@Stifael So to answer your question, by default weathervane is not active you first need to enable it using the dedicated parameters.